### ISSUE TITLE
uMatrix slows down Chrome over time

### ISSUE THREAD

**Issue by paleyellowmoon** [gorhill/uMatrix#351]
I've been using uMatrix on Chrome on Windows 8.x (x86) for about 6 months now, on a dual-core machine with 4GB of memory.

Over the past two months or so, I've noticed that over the course of about a half-hour of continuous web browsing, Chrome begins to slow down significantly, and my CPU begins to peg at 100%.

My web browsing is light to average. I read news sites thru RSS, and maybe about 5% of the pages I read contain an embedded video which I play. I don't play games in my browser, nor do I keep any browser tabs open for a long time.

I have click-to-play enabled in Chrome, so that embedded media ads and such don't automatically play.

I don't use any other programs besides Chrome on my computer when I notice this slowdown.

My uMatrix settings are as follows:

Delete blocked cookies. 
Delete non-blocked session cookies 15 minutes after
Delete local storage content set by blocked hostnames
Clear browser cache every 15 minutes. 
Spoof HTTP referrer string of third-party requests. 
Block all hyperlink auditing attempts. 
Spoof User-Agent string every 5 minutes. 

My permanent rules list contains approximately 1360 rules.

I've twice backed up my settings, and uninstalled and reinstalled uMatrix, but the slowdown problem persists.

Do you have any insight into why the extension is behaving in this way?

I've temporarily reverted to the use of Adblock Plus, which for me doesn't exhibit any of this behavior  even if I've continuously been running Chrome for several days in a row.

**Comment by gorhill** [comment]
> Do you have any insight into why the extension is behaving in this way?

The only way to validate that uMatrix is the cause of the slow down is to profile it when you notice a slow down, otherwise it's just an opinion.

**Comment by gorhill** [comment]
Since you say you can reproduce the issue, you can prove or disprove your feeling that uMatrix is the cause. When you say the "CPU begins to peg at 100%", open the browser's Task Manager and see what process is causing this. If it's a specific web page, open the dev console and use the profiler to create a profile report to substantiate that uMatrix is behind the issue. If the process is uMatrix itself, open the dev console of the extension and profile it to create a profile report to see where in uMatrix the issue is.

**Comment by paleyellowmoon** [comment]
Thanks for following up, Raymond. I apologize for not updating this issue in a more timely manner.

When Chrome+uMatrix pegs my CPU at 100%, I practically can't do anything else, including opening the dev console. The machine becomes very sluggish and unresponsive to normal keyboard or mouse input.

While you're right that I could provide more useful information if I were able to profile uMatrix in the event of a slow down, I'm fairly certain that I've isolated the cause within the crude limits of what I _am_ able to do.

I already mentioned that I removed uMatrix and installed Adblock Plus. I've been using this configuration for the past 5 days, and I haven't encountered a single instance of my machine slowing down.

This evening I uninstalled ABP and re-installed uMatrix. Within 15 minutes of using Chrome to browse the CNN website and stream the American Republican presidential debates, my computer once again hit 100% CPU and slowed down to an absolute crawl.

I shut down the machine, restarted it, uninstalled uMatrix, reinstalled ABP, and went back to streaming the debate. The machine worked fine for the remaining 2.5 hours of the debate, and is still working fine.

I certainly wish I could gather the data that you've described when my machine slows down with uMatrix running, but so far, I'm stymied by the slow machine.
